### ISSUE TITLE
[Question] How to create a global or from scope javascript function ?

### ISSUE THREAD

**Issue by yaronmil** [formio/angular#1108]
I want to reuse a calculate value javascript functin .

I want to be able to call the same function from differant component.

how can it be done ?
is it possible to do so without using the global window objec ?

**Comment by yash-sharma-js** [comment]
you can create a function inside a service and use it anywhere in your application by injecting that service. Angular services are designed to share functionality and data across components, directives, or other services.

**Comment by lane-formio** [comment]
<div>Agent Jeriah Henley linked Freshdesk ticket <a href="https://formhelp.freshdesk.com/a/tickets/53986">53986</a> for this issue.</div><br/>

**Comment by brendanbond** [comment]
Yeah, the only way you're going to be able to "reuse" a function _within_ the form's evaluation context (so, within a calculated value for example) is either by attaching it to the global scope _or_ by injecting it into your form's evaluation context by using the form's `module`, e.g.

```
myForm.module = {
  options: {
    form: {
      evalContext: {
        validatePhone: function(input) {
          return input.match(/^[2-9]\d{2}-\d{3}-\d{4}$/);
        }
      }
    }
  }
}
```
(this way you could reuse the `validatePhone` function across a number of forms
